app/src/main/java/org/akvo/flow/util/SurveyIdGenerator.java

```diff
@@ -33,12 +33,15 @@ public SurveyIdGenerator() {
      * Get the survey id from the survey xml folder
      * The structure can be either surveyId/ or surveyId/folder/
      *
-     * @param folderPath all the folder path expluding the actial fileName
+     * @param entryName all the folder path expluding the actial fileName
      * @return
      */
     @NonNull
-    public String getSurveyIdFromFilePath(@Nullable String folderPath) {
-        String folders[] = folderPath == null ? null : folderPath.split("/");
+    public String getSurveyIdFromFilePath(@Nullable String entryName) {
+        int fileSeparatorPosition = entryName == null? -1 : entryName.lastIndexOf("/");
```

Comment:
I prefer to make sure it also works with more subfolders just in case we never know what may happen, like users adding multiple slashes in the filename, it would then create multiple subfolders.